backend/de.metas.business/src/main/java/de/metas/inout/impl/InOutBL.java

```diff
@@ -804,4 +805,30 @@ public ImmutableSet<I_M_InOut> getNotVoidedNotReversedForOrderId(@NonNull final
 				.build();
 		return inOutDAO.retrieveByQuery(query).collect(ImmutableSet.toImmutableSet());
 	}
+
+
+	@Override
+	public void setShipperId(@NonNull final I_M_InOut inout)
+	{
+		inout.setM_Shipper_ID(ShipperId.toRepoId(findShipperId(inout)));
+	}
+
+	private ShipperId findShipperId(@NonNull final I_M_InOut inout)
+	{
+
+		if (inout.getDropShip_BPartner_ID() > 0 && inout.getDropShip_Location_ID() > 0)
+		{
+			final Optional<ShipperId> deliveryAddressShipperId = bpartnerDAO.getShipperIdByBPLocationId(BPartnerLocationId.ofRepoId(inout.getDropShip_BPartner_ID(), inout.getDropShip_Location_ID()));
+			if (deliveryAddressShipperId.isPresent())
+			{
+				return deliveryAddressShipperId.get(); // we are done
+			}
+		}
```

Comment:
Is the dropship location and partner always set or isn't it set if we use the default ship to address of the order? How do we get the location specific shipper then?

Reply:
if the Dropship location is not set, we get the shipper from dropship partner; this was the requirement.